spec/operators/filter-spec.ts

```diff
@@ -12,11 +12,11 @@ const Observable = Rx.Observable;
 
 /** @test {filter} */
 describe('Observable.prototype.filter', () => {
-  function oddFilter(x) {
+  function oddFilter(x: number | string) {
```

Comment:
should probably just be `number`, same for `isPrime` below... if we have to change the tests, we should. Also I think we should probably use `Number(x)` instead of `+x` in spots because it's a little easier to read. (even tho I am a fan of that trick)